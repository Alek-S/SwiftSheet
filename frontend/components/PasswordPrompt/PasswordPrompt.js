import React, { useState } from 'react';
import styled from 'styled-components';
import defaultStyle, {
	Card,
	ErrorDialog,
	FormCard,
	StyledForm,
	SubmitButton,
} from '../../defaultStyle';

/**
 * @function
 * @name PasswordPrompt
 * Prompts user to provide password
 *
 * @returns {JSX}
 */
export const PasswordPrompt = ({ setPassword, wrongPassword }) => {
	const [formValue, setFormValue] = useState('');

	const handleSubmit = e => {
		e.preventDefault();
		setPassword(formValue);
	};

	return (
		<StyledDiv>
			<PasswordFormCard>
				<StyledForm onSubmit={handleSubmit}>
					<input
						type="password"
						autoComplete="current-password"
						value={formValue}
						onChange={({ target: { value } }) => setFormValue(value)}
						className={wrongPassword ? wrongPassword.toString() : undefined}
						placeholder="Sheet Password"
						minLength="6"
						required
					/>
					<SubmitButton type="submit" value="Submit">
						Submit
					</SubmitButton>
				</StyledForm>
			</PasswordFormCard>

			<ErrorDialog
				className={wrongPassword ? wrongPassword.toString() : undefined}
			>
				⚠️ Incorrect Password Provided
			</ErrorDialog>
		</StyledDiv>
	);
};

const StyledDiv = styled(defaultStyle)`
	padding-top: 6rem;
	background-color: ${props => props.theme.color.background};
	height: calc(100vh - 80px);
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
`;

const PasswordFormCard = styled(FormCard)`
	max-width: 475px;
`;

export { StyledDiv };
