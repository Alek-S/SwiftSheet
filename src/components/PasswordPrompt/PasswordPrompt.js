import React from 'react';
import styled from 'styled-components';
import defaultStyle, { Card, ErrorDialog } from '../../defaultStyle';

/**
 * @function
 * @name PasswordPrompt
 * Prompts user to provide password
 *
 * @returns {JSX}
 */
export const PasswordPrompt = ({ password, setPassword, wrongPassword }) => (
	<StyledDiv>
		<StyledForm>
			<input
				type="password"
				autoComplete="current-password"
				value={password}
				onChange={({ target: { value } }) => setPassword(value)}
				className={wrongPassword && wrongPassword.toString()}
				placeholder="Sheet Password"
			/>
		</StyledForm>

		<ErrorDialog className={wrongPassword}>
			⚠️ Incorrect Password Provided
		</ErrorDialog>
	</StyledDiv>
);

const StyledDiv = styled(defaultStyle)`
	background-color: ${props => props.theme.color.background};
	height: calc(100vh - 55px);
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
`;

const StyledForm = styled(Card)`
	margin-bottom: 1rem;
	padding: 2.5rem 4rem;

	input {
		height: 30px;
		padding: 0.2rem 1.5rem;
		font-size: 1rem;
		font-weight: 300;
		width: 200px;
	}
`;
