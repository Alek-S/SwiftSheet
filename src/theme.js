/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
