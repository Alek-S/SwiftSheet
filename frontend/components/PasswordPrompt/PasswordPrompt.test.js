import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeProvider } from 'styled-components';

import { PasswordPrompt, StyledDiv } from './PasswordPrompt';
import theme from '../../theme';

const setMockPassword = jest.fn();

describe('PasswordPrompt', () => {
	test('matches the snapshot', () => {
		const wrongPassword = true;

		const tree = shallowWithTheme(
			<PasswordPrompt
				password="mockPassword"
				setPassword={setMockPassword}
				wrongPassword={wrongPassword}
			/>,
			theme
		).dive();
		expect(toJson(tree)).toMatchSnapshot();
	});

	test('StyledDiv matches the snapshot', () => {
		const tree = renderer.create(<StyledDiv theme={theme} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('should handle submit of password input', () => {
		const setPassword = jest.fn();

		const ui = render(
			<ThemeProvider theme={theme}>
				<PasswordPrompt setPassword={setPassword} />
			</ThemeProvider>
		);

		const input = ui.getByLabelText('password input');
		fireEvent.change(input, { target: { value: 123 } });

		fireEvent.click(ui.getByText('Submit'));

		expect(setPassword).toHaveBeenCalledWith('123');
	});
});
