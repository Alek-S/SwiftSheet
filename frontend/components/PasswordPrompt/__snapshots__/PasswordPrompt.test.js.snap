// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordPrompt StyledDiv matches the snapshot 1`] = `
.c0 {
  background-color: white;
  color: #404040;
  font-family: 'Quicksand',sans-serif;
  font-size: 0.9em;
  padding-top: 6rem;
  background-color: #fafafa;
  height: calc(100vh - 80px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="defaultStyle-g21vfe-0 c0"
/>
`;

exports[`PasswordPrompt matches the snapshot 1`] = `
<PasswordPrompt__StyledDiv>
  <PasswordPrompt__PasswordFormCard>
    <defaultStyle__StyledForm
      onSubmit={[Function]}
    >
      <input
        aria-label="password input"
        autoComplete="current-password"
        className="true"
        minLength="6"
        onChange={[Function]}
        placeholder="Sheet Password"
        required={true}
        type="password"
        value=""
      />
      <defaultStyle__SubmitButton
        type="submit"
        value="Submit"
      >
        Submit
      </defaultStyle__SubmitButton>
    </defaultStyle__StyledForm>
  </PasswordPrompt__PasswordFormCard>
  <defaultStyle__ErrorDialog
    className="true"
  >
    ⚠️ Incorrect Password Provided
  </defaultStyle__ErrorDialog>
</PasswordPrompt__StyledDiv>
`;
